src/relax/op/op_common.h

```diff
@@ -348,6 +348,10 @@ inline Optional<VDevice> InferBinaryArithOpOutVDevice(const Call& call, const Bl
     }
   };
 
+  if (call->sinfo_args.size() > 0) {
```

Comment:
document the rationale here, we always prefer the device specified in the return statement